homeworks/roman.buzukov_BuzRom/3-async/js/script.js

```diff
@@ -0,0 +1,76 @@
+const containerEl = document.querySelector('[data-container]');
+const listEl = document.querySelector('[data-list]');
+const templateEl = document.querySelector('[data-post-template]');
+const postsEl = [];
+let filteredEl = postsEl;
+const sortEl = document.querySelector('[data-sort]');
+const filterEl = document.querySelector('[data-filter]');
```

Comment:
It looks a little messy for me. Can we sort these variables in terms of meaning? I see here two groups:
* elements;
* arrays of elements;

BTW, when I see `postsEl` and `filteredEl`, I think this is an HTMLElement, but it's an array. Maybe we need to update names for them?

Reply:
Thank you for the good observations!

I get noticed that getElementsByClassName selector is dynamical instead of query selectors.
I'm trying to use it on 'delete' svg, but anyway listener not working before posts loaded.
Have to put listener after fetching posts. Can you help me to solve this problem?

Reply:
You can solve this task in two ways:

1) **Event delegation. (Recommended)**
You can make a global event listener on the post wrapper node. Then, when the user clicks on the delete element, you catch this event and index the post. Then you do your logic. We recommend it because you will have only one listener and less RAM usage.

2) **Separate listeners**
When you have created a "delete" element, you need to add your event listener at that point. Unfortunately, you can't search for elements before you add them to DOM.

Reply:
Please review my solution.

Reply:
Excellent! Thank you!